Move modal outside of the Version component.

artonge opened the pull request:
This is for accessibility, to have the `NcListItem` (`<li>`) as a direct child of the `<ul>`

Before | After
-- | --
![image](https://github.com/nextcloud/server/assets/6653109/86d769be-3434-4e20-bc26-e18e5ac872ce) | ![Screenshot from 2024-01-24 15-48-13](https://github.com/nextcloud/server/assets/6653109/9740f0aa-23f8-4e9c-9390-fdcfbb72b267)

Diff is big, but is mostly white spaces, moved code, and a rewrite in ts of `versions.js`.

JuliaKirschenheuter reviewed (approved):
Thank you Louis!

nfebe reviewed (approved):
Looks good!

ShGKme commented on apps/files_versions/src/views/VersionTab.vue:
```diff
@@ -16,30 +16,37 @@
-		</template>
-		<NcLoadingIcon v-if="loading" slot="loader" class="files-list-viewer__loader" />
-	</VirtualScrolling>
+	<div>
+		<VirtualScrolling :sections="sections"
```
Before | After
---|---
![image](https://github.com/nextcloud/server/assets/25978914/7129395d-0330-4a37-a77e-3c3fe213461a) | ![image](https://github.com/nextcloud/server/assets/25978914/af94bceb-20ae-495a-9b02-b8732db8f9a7)
![image](https://github.com/nextcloud/server/assets/25978914/d23855b0-445b-4a1d-8039-db55a89ff3b4) | ![image](https://github.com/nextcloud/server/assets/25978914/3ffe7b30-83e1-4e72-93dc-edc83cc06dfd)

artonge replied:
Good catch, thanks.
Fixed

artonge committed:
Move modal outside of the Version component.

This is for accessibility, to have the NcListItem (<li>) as a direct child of the <ul>

Signed-off-by: Louis Chemineau <louis@chmn.me>

artonge committed:
Compile assets

Signed-off-by: Louis Chemineau <louis@chmn.me>

susnux commented:
cypress is related and needs fixing

JuliaKirschenheuter commented:
/backport to stable28
